### ISSUE TITLE
Android Compatiblity

### ISSUE THREAD

**Issue by muneebqadir** [oshi/oshi#1533]
Hi, I couldn't find anything on android compatibility so just decided to ask here.
Sorry if I overlooked it but I found it was working with ARM Linux systems

**Comment by mprins** [comment]
I don't know, did you try? 
Oshi might work, but it relies on https://github.com/java-native-access/jna for native code binding which might fail, part of the problem is android does not run java bytecode but dalvik bytecode and binding to native libs might fail and oshi might be using language features not available in your version of android.

may be of interest: https://stackoverflow.com/questions/15765212/android-jna-library

**Comment by dbwiddis** [comment]
JNA is supposed to be Android compatible.   Right now if you tried it in OSHI, you'd get the "Operating System not supported" exception because we haven't added a switch for JNA's `Platform.ANDROID`.  I made a brief attempt to spin up an Android VM to try to test this but didn't get very far. I'm not sure how well an emulator will test this, and it's not been a priority for me to learn how to spin up an emulator just to test it.

It is possible that simply adding that as an option that feeds into OSHI's Linux code will "just work".

It's also possible that some things may break.

Bottom line:  if the Community wants Android support, the community needs to do this testing.

Start by adding `Platform.ANDROID` to the existing `SystemInfo` case statements leading to the Linux branch, and see what works.

**Comment by sbotev5** [comment]
Hi and great to see this being discussed!

First of all, thanks for this amazing library.

I wanted to briefly touch on this library being used on Android devices (as part of the functionality of an Android app).

To start off, I want to ask would this library (assume setup hiccups resolved) provide additional stuff on top of Android SDK programmatic calls already available? 

From a Google search [result](https://www.viralandroid.com/2017/02/how-to-get-android-hardware-and-system-info-programmatically.html) I see the following are available:


`        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                textView.setText(
                        "SERIAL: " + Build.SERIAL + "\n" +
                        "MODEL: " + Build.MODEL + "\n" +
                        "ID: " + Build.ID + "\n" +
                        "Manufacture: " + Build.MANUFACTURER + "\n" +
                        "Brand: " + Build.BRAND + "\n" +
                        "Type: " + Build.TYPE + "\n" +
                        "User: " + Build.USER + "\n" +
                        "BASE: " + Build.VERSION_CODES.BASE + "\n" +
                        "INCREMENTAL: " + Build.VERSION.INCREMENTAL + "\n" +
                        "SDK:  " + Build.VERSION.SDK + "\n" +
                        "BOARD: " + Build.BOARD + "\n" +
                        "BRAND: " + Build.BRAND + "\n" +
                        "HOST: " + Build.HOST + "\n" +
                        "FINGERPRINT: "+Build.FINGERPRINT + "\n" +
                        "Version Code: " + Build.VERSION.RELEASE
                );
            }
        });`

Thanks!

**Comment by dbwiddis** [comment]
I think you’ve got it backwards.  :)  OSHI would internally use native calls like those for the info it reports.  

Most of what I see in that list is covered in ComputerSystem, Baseboard, or Firmware objects.  Oshi also provides much more data like CPU and Memory usage, running processes and threads, disk (or SSD) usage, etc.  

The main goal of OSHI is to provide a cross-platform API so if you are building an app that you want both on Android and other OS’s it would be a good fit.  For an android-only app, there are probably better ways to do it.

**Comment by dbwiddis** [comment]
I was bored today and spent a few hours setting up an Android VM, and used termux for the shell.  Some observations:
 - JNA `Platform` code did not detect this as Android, so it already assumed it was on Linux
 - It got as far as `[INFO] SystemInfoTest - Initializing System...` but then failed with an UnsatisfiedLinkError attempting to dlopen `libc.so.6`.

I think this is a termux problem based on this: https://wiki.termux.com/wiki/Differences_from_Linux

There may be a more direct command line login via ssh that I might try to play around with, but I'm not very hopeful.
